### ISSUE TITLE
python3 ConfigManager broken

### ISSUE THREAD

**Issue by tiarno** [plastex/plastex#65]
Several issues remain with python3 branch. 

    plasTeX/ConfigManager/String.py (self.data not set)

and in python2, we have

    In [1]: from plasTeX.Config import config
    In [2]: sec=config['general']
    In [3]: for option in sec.values():
    ...: print(type(option))
    ...: 
    <class 'plasTeX.ConfigManager.Boolean.BooleanOption'>
    <class 'plasTeX.ConfigManager.Multi.MultiOption'>
    <class 'plasTeX.ConfigManager.String.StringOption'>
    <class 'plasTeX.ConfigManager.Boolean.BooleanOption'>
    <class 'plasTeX.ConfigManager.String.StringOption'>
    <class 'plasTeX.ConfigManager.String.StringOption'>
    <class 'plasTeX.ConfigManager.String.StringOption'>

while in python3, the response is:

    <class 'int'>
    <class 'int'>
    <class 'str'>
    <class 'list'>
    <class 'str'>
    <class 'str'>
    <class 'str'>

--T

**Comment by PatrickMassot** [comment]
First issue can be fixed by adding to ``StringOption.__init__`` a last line  ``self.data = self.data or ''`` but I'd rather understand why. I can follow what happens and when this ``self.data`` can become ``None`` (which is not a sensible data value for a ``UserString``) but I don't understand how it can work with python2 so I'm reluctant to fix it that way.

Second issue is related to how a number of functions return "views" instead of lists in python 3, but in a subtle way. The way the ``ValuesView`` yields values when iterated over conflicts the way the config manager gives options values instead of options instances (see [this comment ](https://github.com/tiarno/plastex/blob/5209229a052f08f1df2e3f63ca8620403ec20fc0/plasTeX/ConfigManager/__init__.py#L227)).
It can be fixed by changing in [this line](https://github.com/tiarno/plastex/blob/5209229a052f08f1df2e3f63ca8620403ec20fc0/plasTeX/ConfigManager/__init__.py#L1068) from ``self.values()`` to ``self.data.values()`` (and I think the ``list`` on this line and the previous ones are useless).

**Comment by kesmit13** [comment]
I would have thought that the UserString.__init__(self, '') line in the StringOption constructor would do what you are describing.

As far as the view (and list) changes go, I don't see any problem with those changes.

**Comment by PatrickMassot** [comment]
You can try it (in master branch, no need for python3). 
```python
In [1]: from plasTeX.ConfigManager.String import StringOption
In [2]: s=StringOption()
In [3]: s.data is None
Out[3]: True
```

The first line in ``StringOption.__init__`` does what you think it does but then ``GenericOption.initialize(self, locals())`` is called. It ends by calling ``GenericOption.setValue`` with a None argument which then calls ``GenericOption.clearValue`` which does ``self.data = None``.

**Comment by kesmit13** [comment]
It looks like the main difference between the UserString implementation in Python 2 and Python 3 is that in Python 2 it uses == for comparisons whereas Python 3 uses cmp(...).  I wonder if there are just enough differences in the implementations that if you used the Python 2 version of the UserString module in Python 3 it would work, and that's why it works in Python 2.  I'd be tempted to just set the data as you mentioned and see what happens.

**Comment by PatrickMassot** [comment]
All this was fixed a long time ago in a229fc385763235b565c93004287b9b878cabc4f and 4e4d922bf27ea9621deb642625396746a10c49e8.
